### DISCUSSION TITLE
connect docker inventree-db (postgres)?

### DISCUSSION THREAD

**Discussion by movelikeriver** [inventree/InvenTree#7214] [Q&A] [answered]
following instruction https://docs.inventree.org/en/latest/start/docker_install/  to setup dev environment, looking for some way to access the DB (postgres).

under `InvenTree/docker` dir, run
`docker compose up -d`
```
[+] Running 5/5
 ✔ Network inventree_default   Created                                                                                                                                 0.0s 
 ✔ Container inventree-db      Started                                                                                                                                12.1s 
 ✔ Container inventree-server  Started                                                                                                                                 0.1s 
 ✔ Container inventree-proxy   Started                                                                                                                                 0.1s 
 ✔ Container inventree-worker  Started            
```
the server is accessible in http://inventree.localhost/index/  (which is expected!)

I'd like to check the data in postgres DB, how to connect to DB?
tried a few command line below, and also browser http://localhost
```
psql -U pguser -h localhost:5432 inventree
```
it's not working.

**Comment by SchrodingersGat** [comment, marked as answer]
@movelikeriver you need to exec `psql` within the database container. Assuming you are using the default config:

```bash
docker exec -it inventree-db psql -U pguser -W inventree
```

This should get you into the postgresql shell.

**Reply by movelikeriver** [reply]
thank you!  `docker exec -it inventree-db psql -U pguser -W inventree` works!

one more related question, how does the `inventree-server` access `inventree-db`?  I was trying to connect DB in `bash`, but it failed 
```
$ docker exec -it inventree-server /bin/sh
```

in docker bash, it can ping `inventree-server` and `db`
```
/home/inventree # ping 0.0.0.0:8000
\PING 0.0.0.0:8000 (0.0.0.0): 56 data bytes
64 bytes from 127.0.0.1: seq=0 ttl=64 time=17.844 ms
64 bytes from 127.0.0.1: seq=1 ttl=64 time=0.205 ms

/home/inventree # ping 0.0.0.0:5432
PING 0.0.0.0:5432 (0.0.0.0): 56 data bytes
64 bytes from 127.0.0.1: seq=0 ttl=64 time=0.176 ms
64 bytes from 127.0.0.1: seq=1 ttl=64 time=0.128 ms
```

however,
```
/home/inventree # psql -U pguser -h 127.0.0.1 -p 5432 -d inventree
psql: error: connection to server at "127.0.0.1", port 5432 failed: Connection refused
	Is the server running on that host and accepting TCP/IP connections?
```

**Reply by SchrodingersGat** [reply]
The host name for the db container (from the server container) should be `inventree-db`.

So the command should be something like:

```bash
/home/inventree # psql -U pguser -h inventree-db -p 5432 -d inventree
```

*Note: I have not tested this locally!*

**Reply by movelikeriver** [reply]
ah, i see. yes, it works. thanks a lot
